--- alpine-conf-2.10.0.orig/setup-apkrepos.in
+++ alpine-conf-2.10.0/setup-apkrepos.in
@@ -91,12 +91,12 @@
 get_alpine_release() {
 	local version=$(cat /etc/alpine-release 2>/dev/null)
 	case "$version" in
-		*_git*) release="edge";;
+		*_git*) release="current";;
 		[0-9]*.[0-9]*.[0-9]*)
 			# release in x.y.z format, cut last digit
 			release=v${version%.[0-9]*};;
 		*)	# fallback to edge
-			release="edge";;
+			release="current";;
 	esac
 }
 
